src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -256,6 +257,24 @@ def prepare_or_run_pipeline(
             processor_args_for_step["base_job_name"] = orchestrator_run_name
             processor_args_for_step["env"] = environment
 
+            # Convert network_config to sagemaker.network.NetworkConfig if present
+            network_config = processor_args_for_step.get("network_config")
+            if network_config and isinstance(network_config, dict):
+                try:
+                    processor_args_for_step["network_config"] = NetworkConfig(**network_config)
+                except TypeError:
+                    # If the network_config passed is not compatible with the NetworkConfig class,
+                    # raise a more informative error.
+                    raise TypeError(
+                        "Expected a sagemaker.network.NetworkConfig compatible object for the network_config argument, "
+                        "but the network_config processor argument is invalid."
+                        "See https://sagemaker.readthedocs.io/en/stable/api/utility/network.html#sagemaker.network.NetworkConfig "
+                        "for more information about the NetworkConfig class."
+                    )
+                except Exception as e:
+                    # Reraise any other error
+                    raise e
```

Comment:
This is redundant - will be raised anyways if you don't catch it

```suggestion
```

Reply:
https://github.com/zenml-io/zenml/pull/1873/commits/4d253ba76d2240c5083e176ddd84d2c1cbe09085